From 7768b91bbbce347cff617ce7d466e261cf638597 Mon Sep 17 00:00:00 2001
From: Patrick Doyle <pdoyle@irobot.com>
Date: Thu, 1 Aug 2019 15:09:40 -0400
Subject: [PATCH] Revert "MIPS: ralink: copy the commandline from the
 devicetree"

This reverts the commit introduced by 0100-prom_fixes.patch.

We don't have a prom on Sundial, and we don't want to bake the device
tree into the kernel (as that means that u-boot cannot modify it).
---
 arch/mips/kernel/head.S |  3 ---
 arch/mips/ralink/of.c   | 21 ++-------------------
 2 files changed, 2 insertions(+), 22 deletions(-)

diff --git a/arch/mips/kernel/head.S b/arch/mips/kernel/head.S
index d702744..d808121 100644
--- a/arch/mips/kernel/head.S
+++ b/arch/mips/kernel/head.S
@@ -85,9 +85,6 @@ EXPORT(__image_cmdline)
 	.fill	0x400
 #endif /* CONFIG_IMAGE_CMDLINE_HACK */
 
-	.ascii  "OWRTDTB:"
-	EXPORT(__image_dtb)
-	.fill   0x4000
 	__REF
 
 NESTED(kernel_entry, 16, sp)			# kernel entry point
diff --git a/arch/mips/ralink/of.c b/arch/mips/ralink/of.c
index 06914ac..832abb2 100644
--- a/arch/mips/ralink/of.c
+++ b/arch/mips/ralink/of.c
@@ -3,7 +3,7 @@
  * under the terms of the GNU General Public License version 2 as published
  * by the Free Software Foundation.
  *
- * Copyright (C) 2008-2014 Imre Kaloz <kaloz@openwrt.org>
+ * Copyright (C) 2008 Imre Kaloz <kaloz@openwrt.org>
  * Copyright (C) 2008-2009 Gabor Juhos <juhosg@openwrt.org>
  * Copyright (C) 2013 John Crispin <john@phrozen.org>
  */
@@ -66,19 +66,6 @@ static int __init early_init_dt_find_memory(unsigned long node,
 	return 0;
 }
 
-static int chosen_dtb;
-
-static int __init early_init_dt_find_chosen(unsigned long node, const char *uname,
-				     int depth, void *data)
-{
-	if (depth == 1 && !strcmp(uname, "chosen"))
-		chosen_dtb = 1;
-
-	return 0;
-}
-
-extern struct boot_param_header __image_dtb;
-
 void __init plat_mem_setup(void)
 {
 	void *dtb = NULL;
@@ -95,11 +82,7 @@ void __init plat_mem_setup(void)
 	else if (__dtb_start != __dtb_end)
 		dtb = (void *)__dtb_start;
 
-	__dt_setup_arch(&__image_dtb);
-
-	of_scan_flat_dt(early_init_dt_find_chosen, NULL);
-	if (chosen_dtb)
-		strlcpy(arcs_cmdline, boot_command_line, COMMAND_LINE_SIZE);
+	__dt_setup_arch(dtb);
 
 	strlcpy(arcs_cmdline, boot_command_line, COMMAND_LINE_SIZE);
 
-- 
2.7.4

